### PR TITLE
Reorder stableCodeFimTemplate.stop order

### PR DESCRIPTION
**Pull request by adawalli**
Some providers have a limited number of STOPs that can be accepted (e.g., NVIDIA NIMs, groq, etc). This means that the ordering of the first 4 default STOPS is important. Starcoder has an interesting issue where it emits <file_sep> as part of it's completion and having this supplied as part of the 4 first stops helps to provide better completions.

## Description

Reordered key 3 and 4 (0 indexed) so that `<file_sep>` makes it's way into even the most restricted of `STOP` lists

## Checklist

- [ ] The base branch of this PR is `dev`, rather than `main`
- [ ] The relevant docs, if any, have been updated or created

## Screenshots

[ "For visual changes, include screenshots." ]

## Testing

[ For new or modified features, provide testing instructions. ]

### CODE REVIEW

File: core/autocomplete/templates.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ const stableCodeFimTemplate: AutocompleteTemplate = {
       "<fim_prefix>",
       "<fim_suffix>",
       "<fim_middle>",
-      "<|endoftext|>",
       "<file_sep>",
+      "<|endoftext|>",
```
Comment (agirault) [inline core/autocomplete/templates.ts:37]: You're updating stablecode which maybe is fine, but did you not mean to update starcoder? https://github.com/continuedev/continue/pull/2057/files#diff-369bc8b51fad8f522e230fb546b9c8ee7fd69650421509eadaf6877320082fd0R137-R138

Comment (adawalli) [reply core/autocomplete/templates.ts:37]: I don't know the history here, but the starcoder template is not currently used - `stable` is used

https://github.com/continuedev/continue/blob/1f20433f36e50bbb344d0de656c7d0b98666a682/core/autocomplete/templates.ts#L314-L330

Comment (adawalli) [reply core/autocomplete/templates.ts:37]: that being said, I should probably update the starcoder template as well in case that line gets uncommented?

Comment (agirault) [reply core/autocomplete/templates.ts:37]: oh, uh. Feedback from @sestinj needed then.

Comment (adawalli) [reply core/autocomplete/templates.ts:37]: I went ahead and updated the `starcoder2FimTemplate` as well to be safe

Comment (sestinj) [reply core/autocomplete/templates.ts:37]: yeah starcoder uses the same template as stablecode so we just have named is stablecode. The changes here look good

**Review by sestinj — APPROVED** [review]

File: core/autocomplete/templates.ts (hunk 1 above)
Comment (agirault) [reply core/autocomplete/templates.ts:37]: ty @sestinj. What is `starcoder2FimTemplate` used for then?

**Pull request merged**